### PR TITLE
Developer sidebar: Support search for Blockly rules/scripts

### PR DESCRIPTION
**Pull request by mherwege**
Searches for 'block', 'blockly', 'Blockly' will return a list of all Blockly rules.

Closes https://github.com/openhab/openhab-webui/issues/1943

### CODE REVIEW

File: bundles/org.openhab.ui/web/src/components/developer/developer-sidebar.vue [hunk 1]
```diff
@@ -493,6 +494,9 @@ export default {
         if (m.configuration.type && m.configuration.type.toLowerCase() === query) {
           return true
         }
+        if (query.indexOf('block') >= 0 && m.configuration.blockSource !== undefined) {
```
Comment (florian-h05) [inline bundles/org.openhab.ui/web/src/components/developer/developer-sidebar.vue:497]:
```suggestion
        if (query.indexOf('block') === 0 && m.configuration.blockSource !== undefined) {
```

This is enough to match `block`, `blockly`, `blockSource` and avoids matching search terms like `oh-block`.

We could even go a step further and only match these three terms above:

```suggestion
        if (['block', 'blockly', 'blocksource'].includes(query) && m.configuration.blockSource !== undefined) {
```

Comment (mherwege) [reply bundles/org.openhab.ui/web/src/components/developer/developer-sidebar.vue]: Thanks. Adjusted.

**Review by florian-h05 — APPROVED** [review]
LGTM, thanks!

**Comment by florian-h05** [comment]
Can you please open a PR for the docs? You‘ll find the edit this page link at the bottom of this page: https://next.openhab.org/docs/tutorial/tips-and-tricks.html#search.

I‘ll merge this when CI passed.

**Pull request merged**